xm-commons-topic/src/main/java/com/icthh/xm/commons/topic/service/TopicConfigurationService.java

```diff
@@ -0,0 +1,116 @@
+
+    private static final String TENANT_NAME = "tenant";
+
+    private AntPathMatcher matcher = new AntPathMatcher();
+    private ObjectMapper ymlMapper = new ObjectMapper(new YAMLFactory());
+
+    private final String configPath;
+
+    private final MessageHandler messageHandler;
+
+    private final DynamicConsumerConfigurationService dynamicConsumerConfigurationService;
+
+    @Getter
+    private Map<String, List<DynamicConsumer>> tenantTopicConsumers = new ConcurrentHashMap<>();
+
+    public TopicConfigurationService(@Value("${spring.application.name}") String appName,
+                                     @Lazy DynamicConsumerConfigurationService dynamicConsumerConfigurationService,
+                                     MessageHandler messageHandler) {
+        this.configPath = "/config/tenants/{tenant}/" + appName + "/topic-consumers.yml";
+        this.dynamicConsumerConfigurationService = dynamicConsumerConfigurationService;
+        this.messageHandler = messageHandler;
+    }
+
+    @Override
+    public void onRefresh(String updatedKey, String config) {
+        refreshConfig(updatedKey, config);
+    }
+
+    @Override
+    public boolean isListeningConfiguration(String updatedKey) {
+        return matcher.match(configPath, updatedKey);
+    }
+
+    @Override
+    public void onInit(String configKey, String configValue) {
+        if (isListeningConfiguration(configKey)) {
+            refreshConfig(configKey, configValue);
+        }
+    }
+
+    @Override
+    public List<DynamicConsumer> getDynamicConsumers(String tenantKey) {
+        if (tenantTopicConsumers.containsKey(tenantKey)) {
+            return tenantTopicConsumers.get(tenantKey);
+        }
+
+        return new ArrayList<>();
+    }
+
+    private void refreshConfig(String updatedKey, String config) {
+        String tenantKey = extractTenant(updatedKey);
+
+        if (StringUtils.isEmpty(config)) {
+            tenantTopicConsumers.remove(tenantKey);
+        } else {
+            readSpec(updatedKey, config).ifPresentOrElse(spec -> {
+                List<TopicConfig> forUpdate = spec.getTopics();
+                List<DynamicConsumer> dynamicConsumers = forUpdate.stream()
+                        .map(this::createDynamicConsumer)
+                        .collect(Collectors.toList());
+                tenantTopicConsumers.put(tenantKey, dynamicConsumers);
+            }, () -> log.warn("Skip processing of configuration: [{}]. Specification is null", updatedKey));
+        }
+
+        dynamicConsumerConfigurationService.refreshDynamicConsumers(tenantKey);
+    }
+
+    private String extractTenant(final String updatedKey) {
+        return matcher.extractUriTemplateVariables(configPath, updatedKey).get(TENANT_NAME);
+    }
+
+    private Optional<TopicConsumersSpec> readSpec(String updatedKey, String config) {
+        TopicConsumersSpec spec = null;
```

Comment:
TopicConsumersSpec spec = ymlMapper.readValue(config, TopicConsumersSpec.class);
return Optaion.of(spec);